core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -351,4 +353,26 @@ default EventExecutor defaultSubscriberExecutor() {
      * on a closed or aborted stream has no effect.
      */
     void abort(Throwable cause);
+
+    /**
+     * Filters values emitted by this {@link StreamMessage}.
+     * If the {@link Predicate} test succeeds, the value is emitted.
+     * If the {@link Predicate} test fails, the value is ignored and a request of {@code 1} is made to upstream.
+     */
+    default StreamMessage<T> filter(Predicate<? super T> predicate) {
+        requireNonNull(predicate, "predicate");
+        return FuseableStreamMessage.of(this, predicate);
+    }
+
+    /**
+     * Transforms values emitted by this {@link StreamMessage} by applying the specified {@link Function}.
+     * As per
+     * <a href="https://github.com/reactive-streams/reactive-streams-jvm#2.13">
+     * Reactive Streams Specification 2.13</a>, the specified {@link Function} should not return
+     * a {@code null} value.
+     */
+    default <U> StreamMessage<U> map(Function<? super T, ? extends U> function) {
+        requireNonNull(function, "function");
+        return FuseableStreamMessage.of(this, function);
```

Comment:
Should we optimize this a little bit, like returning `this` when `function == Function.identity()`?

Reply:
>  Is this the beginning of `ikstream`? 😆

It should be `SongStream`. 🤣 We are gonna add other useful stream operators such as `flatMap` and `concatMap`.